### DISCUSSION TITLE
GraphQL Dependency Issue with NPM Upgrade

### DISCUSSION THREAD

**Discussion by BPowell76** [requarks/wiki#7369] [Error / Bug Report] [answered]
Not sure if this is considered a bug or error, but wanted to bring this up in the event it wasn't already discussed. When running `npm upgrade` with npm@10.8.3 (latest) I get the following dependency issues with GraphQL and GraphQL-Subscriptions:

```
npm error code ERESOLVE
npm error ERESOLVE unable to resolve dependency tree
npm error
npm error While resolving: wiki@2.5.303
npm error Found: graphql@15.3.0
npm error node_modules/graphql
npm error   graphql@"15.3.0" from the root project
npm error
npm error Could not resolve dependency:
npm error peer graphql@"^0.10.5 || ^0.11.3 || ^0.12.0 || ^0.13.0 || ^14.0.0" from graphql-subscriptions@1.1.0
npm error node_modules/graphql-subscriptions
npm error   graphql-subscriptions@"1.1.0" from the root project
npm error
npm error Fix the upstream dependency conflict, or retry
npm error this command with --force or --legacy-peer-deps
npm error to accept an incorrect (and potentially broken) dependency resolution.
```

Wiki.js version 2.5.303 works fine (been using it for a few months now), but I don't think these two packages work together if I am understanding this error message. I didn't use --force since the site is working and didn't want to break anything by trying to upgrade to newer packages.

**Comment by NGPixel** [comment, marked as answer]
Don't run `npm upgrade`. You're going to end up with a broken installation.

Many dependencies have breaking changes.

**Reply by BPowell76** [reply]
Thanks for the heads up. I'll adjust my cron jobs accordingly. Upgrading Node to the newest available LTS versions is fine or should that be monitored more closely too?

**Reply by NGPixel** [reply]
That's fine.

**Reply by marciovinicius1** [reply]
> That's fine.

i have the same problem and i can't run `npm install`. Can you help me?

**Reply by NGPixel** [reply]
**Read the [instructions](https://docs.requarks.io/install)**, you shouldn't be running any npm commands...
